### ISSUE TITLE
Snowflake copy destinations should execute COPY commands in parallel

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#9087]
## Tell us about the problem you're trying to solve
Currently, destination-snowflake's copy modes will generate multiple files on S3 or GCS, and then for each of those files, execute a COPY command _in serial_. We should run those commands in parallel to be more time-efficient.

Additionally, the COPY command can actually accept [up to 1,000 files](https://docs.snowflake.com/en/user-guide/data-load-considerations-load.html#lists-of-files). `destination-snowflake` should take advantage of that capability.

## Describe the solution you’d like
* After https://github.com/airbytehq/airbyte/issues/8820 is completed, update [`S3StreamCopier`](https://github.com/airbytehq/airbyte/blob/ff5648aa953789819121dca0a694b242438f40d5/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/s3/S3StreamCopier.java#L150-L153) and [`GcsStreamCopier`](https://github.com/airbytehq/airbyte/blob/ff5648aa953789819121dca0a694b242438f40d5/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/gcs/GcsStreamCopier.java#L140-L142) to run `copyS3CsvFileIntoTable` / `copyGcsCsvFileIntoTable` in parallel
* Make `SnowflakeS3StreamCopier` and `GcsStreamCopier` override `copyStagingFileToTemporaryTable` to actually process 1,000 files at a time, and have `copyS3CsvFileIntoTable` / `copyGcsCsvFileIntoTable` throw RuntimeException. (see `RedshiftStreamCopier`, which is similar).
  * Write unit tests to verify this behavior - again, see `RedshiftStreamCopierTest`'s usage of `verify(db).execute(...)`

## Are you willing to submit a PR?
👍 😺

**Comment by joshuataylor** [comment]
That'd be great, but there should also be a configuration set for max size of tmp directory, in case the download of source is faster than the upload to Snowflake. This way the disk doesn't fill up with tmp files whilst uploading to SF if the outbound connection speed is congested.

**Comment by edgao** [comment]
ooh, yeah that's a great idea! Could you make a separate issue to track that? I suspect it would be a bit separate from the features in this one

also, to double-check - you're referring to the internal staging mode? The S3 and GCS staging methods shouldn't (?) run into disk size limits
